### ISSUE TITLE
MA0074 Raised for replace on netstandard2.0

### ISSUE THREAD

**Issue by Pretasoc** [meziantou/Meziantou.Analyzer#547]
### Version of the Meziantou.Analyzer NuGet package

2.0.54

### Rule Identifier

MA0074

### Target Framework

netstandard2.0

### C# Language version

C# 9

### Description

In a project that targets `netstandard2.0`, `net472`, `netcoreapp3.1` and `net6.0`. This warning is issued for the `string.Replace` Method in `netstandard2.0`. However there is no overload taking a `StringComparion` in that framework.

### Reproduction Steps

Minimal code:

```c#
#if NETCOREAPP3_1_OR_GREATER
"FooBar".Replace("Bar", "Baz", StringComparison.Ordinal);
#else
"FooBar".Replace("Bar", "Baz")
#endif
```

### Other information

The documentation of `string.Replace(string, string, StringComparison)` on [learn.microsoft.com](https://learn.microsoft.com/en-us/dotnet/api/system.string.replace?view=net-7.0#system-string-replace(system-string-system-string-system-stringcomparison))

**Comment by Pretasoc** [comment]
Turns out, it was a visual studio bug. When building from command line, the error was not reported

**Comment by meziantou** [comment]
I cannot reproduce on my side:

````
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFrameworks>net8.0;netstandard2.0;net472;netcoreapp3.1</TargetFrameworks>
    <LangVersion>11</LangVersion>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Meziantou.Analyzer" Version="2.0.60">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
  </ItemGroup>

</Project>
````

````
using System;

// Reported as expected
_ = "FooBar".Replace("Bar", "Baz");

// No report
#if NETCOREAPP3_1_OR_GREATER
_ = "FooBar".Replace("Bar", "Baz", StringComparison.Ordinal);
#else
_ = "FooBar".Replace("Bar", "Baz");
#endif
````

You can see that it reports the first `Replace` on net8 and netcoreapp3.1

<img width="459" alt="image" src="https://github.com/meziantou/Meziantou.Analyzer/assets/509220/62bd2ad1-624e-4861-a6e4-685b5245bd6a">
